src/lib/layer.js

```diff
@@ -446,7 +446,10 @@ export default class Layer {
     settings.polygonOffset = offsets;
 
     // Call subclass lifecycle method
-    this.draw({moduleParameters, uniforms, settings});
+    withParameters(this.context.gl,
```

Comment:
There should be many use cases for a layer `settings` prop (`props.parameters` or `props.glParameters`).  An editing app that I work on now wants to control blend modes per layer, and this parameter would handle that nicely with our stock layers.

Exposing this gives a ton of flexibility to apps, at a very low API cost for deck.gl (luma.gl already does the heavy lifting of allowing parameters to be specified as keys in objects). 

No objections to changing to `glParameters` although it would be (partly) inconsistent with luma.gl, where the the new `Model.draw({uniforms, ..., settings})` function takes a `settings` field. That said, the global luma.gl **function** is called `setParameters`.

I'd probably vote for `parameters` and if so, maybe update `Model.draw({settings})` to take `Model.draw({parameters})` as well?

@1chandu @shaojingli